projects/igniteui-angular/src/lib/drop-down/drop-down.component.ts

```diff
@@ -524,18 +525,22 @@ export class IgxDropDownComponent extends IgxDropDownBaseDirective implements ID
                 index: newSelection.index
             } as IgxDropDownItemBaseDirective;
         }
-        const args: ISelectionEventArgs = { oldSelection, newSelection, cancel: false, owner: this };
-        this.selectionChanging.emit(args);
+        let args: ISelectionEventArgs;
+        if (emit) {
+            args = { oldSelection, newSelection, cancel: false, owner: this };
+            this.selectionChanging.emit(args);
+        }
```

Comment:
Honestly, not much to be gained in perf by skipping the args creation IMO, the code was fine as it was, just a single if around the actual emit should do:
```suggestion
        const args: ISelectionEventArgs = { oldSelection, newSelection, cancel: false, owner: this };

        if (emit) {
            this.selectionChanging.emit(args);
        }
```
and then `if (!args.cancel)` and the rest of the logic can flow as it was :)